static/app/utils/url/stripURLOrigin.tsx

```diff
@@ -0,0 +1,19 @@
+import {safeURL} from 'sentry/utils/url/safeURL';
+
+/**
+ * Accept a url like:
+ * `https://example.com/path/name?query=params#hash` and return
+ * `/path/name?query=params#hash`
+ */
+export default function stripURLOrigin(url: string): string {
+  const parsedUrl = safeURL(url);
+
+  if (!parsedUrl) {
+    return url;
+  }
+
+  // We cannot do something like parsedUrl.pathname + parsedUrl.search + parsedUrl.hash
+  // as the parsed representation of the url may not exactly match the string url argument.
+  // See more detailed explanation in https://github.com/getsentry/sentry/pull/62839#discussion_r1446450773
+  return url.replace(parsedUrl.origin, '');
```

Comment:
This needs to be

```suggestion
  return parsedUrl.href.replace(parsedUrl.origin, '');
```

Reply:
Isn't slice better than replacing?

Reply:
The previous implementation was so I think this is correct.
```
export default function stripOrigin(url: string) {
  return url.replace(parseUrl(url)?.origin ?? '', '');
}
```

Slice suggestion seems good as we can bypass the regexp engine